### ISSUE TITLE
[ToC vertical] keyboard navigation updates

### ISSUE THREAD

**Issue by oliviaflory** [carbon-design-system/carbon-for-ibm-dotcom#10322]
### User story
We need to update the functional specs for the Vertical ToC to reflect the keyboard navigation recommendations from the Accessibility Guild.

### Notes from the review
TOCs provide a few different functions.

1. The first it to provide the user with the outline of the content (provide context).
2. The second is to orient the user to their current position in a long document (which happens by the ‘current’ TOCs item being indicated visually-- and hopefully programmatically).
3. The third is to provide a means of interacting/navigating to TOCs entries. 
   4. For this, the user activates an entry, which re-orients the user to that location in the same large content. We discussed whether focus should remain in the TOC after activitation or move with the viewport re-orientation. Conclusion was that it was much better for the keyboard interaction (focus) to match the repositioning of the content in the viewport. It is not necessary for a focus indicator to appear around the heading that ‘has focus’; but providing each TOC entry with a tabindex=“-1” and then programmatically putting the focus there after activiation is desirable.

### To do
- Vertical ToC should be a navigation region so keyboard users can navigate in and out of the region
- Once user activates an entry the page will scroll to that entry and the focus will be on the heading selected (not the toc)
- Identify what key the user can use to jump back into the toc.

**Comment by RichKummer** [comment]
## Design crit 05-25-23

- Using the tab region is promising for ToC. 
- Look at `tab` panel flow. The only difference here is that tab is switching content, unlike here where we're navigating the view.
- Ideally, the view is moved as the keyboard user is tabbing through the ToC. Once they hit `enter`, the focus shifts to the heading in the corresponding section

**Comment by RichKummer** [comment]
**[Figma v2 specs](https://www.figma.com/file/rUQ0SVSdyC6uumEg6hQPS3/Table-of-contents-v2?type=design&node-id=1%3A2&t=dQ6FqX8iSqCvzIN7-1)** (keyboard navigation only)

**Comment by RichKummer** [comment]
## Ram discussion 06/12/23

- Only the current selected element is tabbable, and arrow keys navigate. Tab is reserved for moving from the ToC region to the next piece of content on the page.
    - In this example, use arrow right/left (for tabs). Arrow keys should navigate the ToC items inside the ToC region.
    - In the example, when using the right arrow on the last item, the user is brought to the first one. The same works for the first item – the tabs form a loop. We can do the same for ToC.
        - `enter` key is used to navigate to section.
        - “Automatic activation” vs “manual activation” – automatic changes the view based on the selection _before_ the user confirms. Manual changes the view only when the user clicks `enter`. A ToC most commonly uses manual activation.
        - This may have an impact on different users based on disability. Based on screen reader, as long as the label is identified programmatically we’re fine.
        - Using automatic activation for the ToC is not a common pattern to update.
        - Expected behavior – tabbing from ToC will send user to the first interactive element in the current page view.
        - Unlike tab, it would need to behave like a typical tab panel or leave as-is.
            - It would be confusing to blind user and sighted user (and cognitively disabled). They would wonder why focus has gone there.

**Comment by RichKummer** [comment]
## Ram discussion 06/13/23
**Question:** Is there a pattern for returning to the `ToC` when inside page content?
> Generally, no. There's two types of users to consider here: 
Screen reader users can navigate from whatever section they are viewing using quick navigation for regions.
Keyboard users: There's no method to return to `ToC` quickly.
It’s common to include a skip-back link for _footnotes_, but have not come across the same for a table of contents component.
